bundles/sirix-core/src/main/java/org/sirix/utils/IntToObjectMap.java

```diff
@@ -619,7 +634,7 @@ public boolean equals(Object o) {
 
       T v1 = this.get(key);
       T v2 = that.get(key);
-      if ((v1 == null && v2 != null) || (v1 != null && v2 == null) || (!v1.equals(v2))) {
+      if ((v1 == null && v2 != null) || (v1 != null && v2 == null) || (v1!= null && !v1.equals(v2))) {
```

Comment:
we also may be able to use `if (!Object.equals(v1, v2)) { return false };`, but I'd have to check the implementation ;-)

Reply:
I'll keep that in mind